src/Microsoft.OpenApi.Readers/OpenApiYamlReader.cs

```diff
@@ -47,14 +47,15 @@ public async Task<ReadResult> ReadAsync(TextReader input,
         /// <inheritdoc/>
         public async Task<ReadFragmentResult<T>> ReadFragmentAsync<T>(TextReader input,
                                                                       OpenApiSpecVersion version,
-                                                                      OpenApiReaderSettings settings = null) where T : IOpenApiElement
+                                                                      OpenApiReaderSettings settings = null,
+                                                                      CancellationToken token = default) where T : IOpenApiElement
         {
             JsonNode jsonNode;
 
             // Parse the YAML
             try
             {
-                jsonNode = LoadJsonNodesFromYamlDocument(input);
+                jsonNode = await Task.Run(() => LoadJsonNodesFromYamlDocument(input));
```

Comment:
why are we adding the task run here?

Reply:
The caller is async here https://github.com/microsoft/OpenAPI.NET/blob/f07ff446a2833e09a6617ae80d534bab4b56fd68/src/Microsoft.OpenApi/Reader/OpenApiModelFactory.cs#L147

And the interface method's equivalent in the JsonReader class is also async https://github.com/microsoft/OpenAPI.NET/blob/f07ff446a2833e09a6617ae80d534bab4b56fd68/src/Microsoft.OpenApi/Reader/OpenApiJsonReader.cs#L124-L143

If I change this method signature to be synchronous, I'll also need to update this JSON content to be read synchronously:
https://github.com/microsoft/OpenAPI.NET/blob/f07ff446a2833e09a6617ae80d534bab4b56fd68/src/Microsoft.OpenApi/Reader/OpenApiJsonReader.cs#L189